packages/site/src/hooks/usePageValues.tsx

```diff
@@ -0,0 +1,160 @@
+import { useMemo, useState } from "react";
+import { InputText, Switch, Text } from "@jobber/components";
+import { ValueStateInternals } from "../types/services";
+import { ContentExport } from "../types/content";
+import { SelectWithOptions } from "../components/SelectWithOptions";
+
+/**
+ * This is doing a few different things.
+ *
+ * 1. Any change to the PropList will call updateValue, which will update the values state.
+ * 2. The stateValues are memoized to prevent re-renders and modified to the DataList format.
+ * 3. Any value that is a straight function is converted to a real function so it can be used in the actual component.
+ * 4. There are two utility functions that may move out of here:
+ *   - isFunction: Checks if a string is a function
+ *  - determineListComponent: Determines what component to use based on the type of the prop
+ * @param meta Content Meta
+ * @returns Values to be used on the ContentView page
+ */
+export const usePageValues = (meta: ContentExport) => {
+  const [values, setValues] = useState(meta.component.defaultProps);
+
+  const updateValue = (key: string, value: string | number | boolean) => {
+    if (key) {
+      setValues(prev => ({
+        ...prev,
+        [key]: value as string,
+      }));
+    }
+  };
+
+  const stateValues = useMemo(() => {
+    return meta.props.map(propList => {
+      return {
+        name: propList.displayName,
+        props: Object.keys(propList.props).map((key, index) => {
+          const prop = propList.props[key];
+
+          return {
+            id: index,
+            key,
+            required: prop?.required ? "*" : "",
+            description: prop?.description,
+            component: determineListComponent({
+              key,
+              type: prop?.type.name || "",
+              placeholder: key,
+              value: values[key] as string,
+              updateValue,
+            }),
+          };
+        }),
+      };
+    });
+  }, [values]);
+
+  const valuesWithFunctions: Record<
+    string,
+    string | number | boolean | undefined
+  > = {};
+
+  for (const i in values) {
+    if (Object.hasOwnProperty.call(values, i)) {
+      valuesWithFunctions[i] = isFunction(values[i] as string)
+        ? // eslint-disable-next-line no-new-func
+          new Function(`return ${values[i]}`)()
+        : values[i];
+    }
+  }
+
+  return {
+    values,
+    updateValue,
+    stateValues,
+    stateValueWithFunction:
+      valuesWithFunctions as unknown as ValueStateInternals,
+  };
+};
+
+const determineListComponent = ({
```

Comment:
This is part of the datalist. Depending on the type of the current prop, we render a different component for interaction. Strings get text inputs, numbers get number inputs, booleans get a switch, Functions get a multiline text input (probably an opportunity for consolidation on props there), and a Select if the type is a series of options